rs/crypto/ecdsa_secp256r1/src/lib.rs

```diff
@@ -399,4 +579,39 @@ impl PublicKey {
 
         self.key.verify_prehash(digest, &signature).is_ok()
     }
+
+    /// Derive a public key from this public key using a derivation path
+    ///
+    /// This is the same derivation system used by the Internet Computer when
+    /// deriving subkeys for threshold ECDSA with secp256r1
+    ///
+    pub fn derive_subkey(&self, derivation_path: &DerivationPath) -> (Self, [u8; 32]) {
+        let chain_code = [0u8; 32];
+        self.derive_subkey_with_chain_code(derivation_path, &chain_code)
+    }
+
+    /// Derive a public key from this public key using a derivation path
+    /// and chain code
+    ///
+    /// This is the same derivation system used by the Internet Computer when
+    /// deriving subkeys for threshold ECDSA with secp256r1
+    ///
+    /// This derivation matches SLIP-10
+    pub fn derive_subkey_with_chain_code(
+        &self,
+        derivation_path: &DerivationPath,
+        chain_code: &[u8; 32],
+    ) -> (Self, [u8; 32]) {
+        let public_key: AffinePoint = *self.key.as_affine();
+        let (pt, _offset, chain_code) = derivation_path.derive_offset(public_key, chain_code);
+
+        let derived_key = Self {
+            key: p256::PublicKey::from(
+                p256::PublicKey::from_affine(pt).expect("Derived point is valid"),
+            )
+            .into(),
```

Comment:
It seems one conversion layer is to itself, so it should be obsolete. Also, how about being explicit (instead of doing `into`). As a result, I suggest we do
```
            key: p256::ecdsa::VerifyingKey::from(
                p256::PublicKey::from_affine(pt).expect("Derived point is valid"),
            ),
```

Reply:
Done